core/src/main/java/org/apache/iceberg/BaseIncrementalChangelogScan.java

```diff
@@ -165,6 +170,13 @@ public CloseableIterable<ChangelogScanTask> apply(
                     context.residuals());
 
               case DELETED:
+                // TODO: this is currently not accurate, rather than counting delete files, we
+                // should use/have a dimensional counter by "task type" here
```

Comment:
I think what we report here will function as a contract, so we should get it right the first time around. I'm not convinced that we necessarily need to use a dimensional counter, but if that's what you want to do I'd prefer that to reusing a different counter temporarily.